### PR TITLE
Generate an RSS feed

### PR DESCRIPTION
**Pull request by gfontenot**
See commit messages for deets

### CODE REVIEW

**Comment by gfontenot** [comment]
cc @pbrisbin since he's doing a lot of this manually right now

File: site.hs [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
 
 main :: IO ()
 main = hakyll $ do
+    match allPosts $ do
+        route $ setExtension "" `composeRoutes` indexedRoute
+        compile $ pandocCompiler
+            >>= saveSnapshot "content"
+            >>= loadAndApplyTemplate "templates/post.html"    postCtx
+            >>= loadAndApplyTemplate "templates/default.html" postCtx
+            >>= replaceIndexLinks
```
Comment (sidraval) [inline site.hs]: Can you used a nested `do` here?

File: site.hs [hunk 2]
```diff
@@ -20,12 +28,21 @@ main = hakyll $ do
...
+                    siteTitle           -- title
+                    "Gordon Fontenot"   -- description
+                    "Gordon Fontenot"   -- author name
+                    "gordon@fonten.io"  -- author email
+                    siteHost            -- root
+
+            renderAtom feedConfig feedItemCtx posts
+                >>= replaceIndexURLs siteHost
+                >>= repairExternalURLs siteHost
+                >>= replaceRelativeURLs siteHost
```
Comment (sidraval) [inline site.hs]: Perhaps nested `do` here as well?

File: src/URLHelper.hs [hunk 3]
```diff
@@ -30,6 +32,17 @@ replaceIndexLinks = replace "href=\"/[^\"]*/index.html" P.takeDirectory
...
+-- | Replaces @/images/foo@ with @<host>/inages/foo@ for the given host
+replaceRelativeURLs :: String -> Item String -> Compiler (Item String)
+replaceRelativeURLs host = replace "=\"/.*\"" prependHost
+
+  where
+    prependHost = ("=\"" <> host </>) . (drop 3)
+
+-- | Repairs urls that have incorrectly had the given host prepended to them
+repairExternalURLs :: String -> Item String -> Compiler (Item String)
+repairExternalURLs host = replace (host <> "http") (drop (length host))
```
Comment (sidraval) [inline src/URLHelper.hs]: Could probably write this point free and lose a bunch of parens if you prefer

Comment (gfontenot) [inline src/URLHelper.hs]: the type of `replace` actually makes that _super_ difficult. I'm already using an implicit argument for the `Item String`, making it entirely point free ends up being pretty gross. I guess I could do `(drop $ length host)` instead though.

File: site.hs (hunk 2 above)
Comment (gfontenot) [inline site.hs]: I've tried that and it didn't look as nice to me. I'll try it again in a different PR.

Comment (sidraval) [inline site.hs]: Alternatively you could define the `renderAtom` bit as a separate function

Comment (gfontenot) [inline site.hs]: Yeah, I think I'm eventually going to do more abstraction for these compilers. Right now, I kinda like the pipeline for explicitness and to help me understand what's going on while I fully grok how this works.

**Pull request merged**